implementations/rust/ockam/ockam_command/src/lib.rs

```diff
@@ -297,13 +296,16 @@ pub fn run() {
         .map(replace_hyphen_with_stdin)
         .collect::<Vec<_>>();
 
-    let command = OckamCommand::parse_from(input);
-
-    if !command.global_args.test_argument_parser {
-        check_if_an_upgrade_is_available();
-    }
+    match OckamCommand::try_parse_from(input) {
+        Ok(command) => {
+            if !command.global_args.test_argument_parser {
+                check_if_an_upgrade_is_available();
+            }
 
-    command.run();
+            command.run();
+        }
+        Err(help) => show_help(help),
```

Comment:
Given that the `show_help` function is pretty short and it's only called here, I think it would be better to remove it and paste the code here.